src/main.ts

```diff
@@ -55,7 +56,23 @@ export async function run(argv = process.argv.slice(2), options?: Interfaces.Loa
   const cmd = config.findCommand(id)
   if (!cmd) {
     const topic = config.findTopic(id)
-    if (topic) return config.runCommand('help', [id])
+    if (topic) {
+      const subCommands = config.commands.filter(c => (c.id).startsWith(id))
+      const argsOfSubCommands = subCommands.reduce((x, y) => x.concat(y.args), [] as Arg[])
+
+      if (argsOfSubCommands.length > 0 || argvSlice[0].startsWith('--')) return config.runCommand('help', [id])
+
+      const extraBits = []
+      for (const arg of argvSlice) {
+        if (arg.startsWith('--')) {
+          break
+        } else {
+          extraBits.push(arg)
+        }
+      }
+      id = [id, ...extraBits].join(':')
```

Comment:
So basically you are trying to find arguments before the first defined flag and create a new command id. Shouldn't this be what `findCommand` does?

This is hard for me to follow. Can you add some more info in some comments, tests, or the PR description? Maybe we should meet to discuss.